### PR TITLE
feat: update Flex component and use GitHub markdown style for documentation

### PR DESCRIPTION
**Pull request by cheton**
# Flex

Flex is [Box](box) with `display: flex` and comes with helpful style shorthand props.

## Import

```js
import Flex from '@trendmicro/react-styled-core/Flex';
// or
import { Flex } from '@trendmicro/react-styled-core';
```

## Usage

Flex provides the following style shorthand props:

| Shorthand Prop | Style Prop | Description |
| :------------- | :--------- | :----------- |
| direction | flexDirection | The `flex-direction` property sets how flex items are placed in the flex container |
| wrap | flexWrap | The `flex-wrap` property sets whether flex items are forced onto one line or can wrap onto multiple lines |
| align | alignItems | The `align-items` property sets the `align-self` value on all direct children as a group |
| justify | justifyContent | The `justify-content` property defines how the browser distributes space between and around content items along the main axis of a flex container |

### Flex items

You can use the `flex` property to set how a flex item will grow or shrink to fit the available space in its flex container. It is a shorthand for `flex-grow`, `flex-shrink`, and `flex-basis`. For most purposes, you should set `flex` to one of the following values: `auto`, `initial`, `none`, or a positive unitless number.

##### `initial`
The item is sized according to its width and height properties. It shrinks to its minimum size to fit the container, but does not grow to absorb any extra free space in the flex container. This is equivalent to setting `flex: 0 1 auto`.

##### `auto`
The item is sized according to its width and height properties, but grows to absorb any extra free space in the flex container, and shrinks to its minimum size to fit the container. This is equivalent to setting `flex: 1 1 auto`.

##### `none`
The item is sized according to its width and height properties. It is fully inflexible: it neither shrinks nor grows in relation to the flex container. This is equivalent to setting `flex: 0 0 auto`.

To see the effect of these values, try resizing the flex containers below:

image

```jsx
function Example() {
  const FlexItem = (props) => (
    <Box p=".75rem" mr="1em" mb="1rem" {...props} />
  );

  return (
    <>
      <Flex bg="gray.20" p="1rem" pb={0} pr={0} mb="1rem">
        <FlexItem bg="blue.50" flex="auto">auto</FlexItem>
        <FlexItem bg="blue.50" flex="auto">auto</FlexItem>
        <FlexItem bg="blue.50" flex="auto">auto</FlexItem>
      </Flex>
      <Flex bg="gray.20" p="1rem" pb={0} pr={0} mb="1rem">
        <FlexItem bg="blue.50" flex="auto">auto</FlexItem>
        <FlexItem bg="blue.50" flex="initial">initial</FlexItem>
        <FlexItem bg="blue.50" flex="initial">initial</FlexItem>
      </Flex>
      <Flex bg="gray.20" p="1rem" pb={0} pr={0} mb="1rem">
        <FlexItem bg="blue.50" flex="auto">auto</FlexItem>
        <FlexItem bg="blue.50" flex="auto">auto</FlexItem>
        <FlexItem bg="blue.50" flex="none">none</FlexItem>
      </Flex>
      <Flex bg="gray.20" p="1rem" pb={0} pr={0} mb="1rem">
        <FlexItem bg="blue.50" flex="initial">initial</FlexItem>
        <FlexItem bg="blue.50" flex="none">none</FlexItem>
        <FlexItem bg="blue.50" flex="none">none</FlexItem>
      </Flex>
      <Flex bg="gray.20" p="1rem" pb={0} pr={0}>
        <FlexItem bg="blue.50" flex={4}>4</FlexItem>
        <FlexItem bg="blue.50" flex={2}>2</FlexItem>
        <FlexItem bg="blue.50" flex={1}>1</FlexItem>
      </Flex>
    </>
  );
}
```

### The `direction` (or `flexDirection`) property

image

```jsx
function Example() {
  const FlexItem = (props) => (
    <Box p=".75rem" mr="1em" mb="1rem" {...props} />
  );

  const [value, setValue] = React.useState('row');

  return (
    <>
      <Box mb="1rem">
        <Button onClick={() => setValue('row')}>row</Button>
        <Button onClick={() => setValue('row-reverse')}>row-reverse</Button>
        <Button onClick={() => setValue('column')}>column</Button>
        <Button onClick={() => setValue('column-reverse')}>column-reverse</Button>
      </Box>
      <pre>
        {`<Flex direction="${value}">`}
      </pre>
      <Flex direction={value} bg="gray.20" p="1rem" pb={0} pr={0}>
        <FlexItem bg="blue.50" flex="auto">One</FlexItem>
        <FlexItem bg="blue.50" flex="auto">Two</FlexItem>
        <FlexItem bg="blue.50" flex="auto">Three</FlexItem>
      </Flex>
    </>
  );
}
```

### The `wrap` (or `flexWrap`) property

image

```jsx
function Example() {
  const FlexItem = (props) => (
    <Box p=".75rem" mr="1rem" mb="1rem" {...props} />
  );

  const [value, setValue] = React.useState('nowrap');

  return (
    <>
      <Box mb="1rem">
        <Button onClick={() => setValue('nowrap')}>nowrap</Button>
        <Button onClick={() => setValue('wrap')}>wrap</Button>
        <Button onClick={() => setValue('wrap-reverse')}>wrap-reverse</Button>
      </Box>
      <pre>
        {`<Flex wrap="${value}">`}
      </pre>
      <Flex wrap={value} width={480} bg="gray.20" p="1rem" pb={0} pr={0}>
        <FlexItem bg="blue.50" flex="auto" minWidth={120}>One</FlexItem>
        <FlexItem bg="blue.50" flex="auto" minWidth={120}>Two</FlexItem>
        <FlexItem bg="blue.50" flex="auto" minWidth={120}>Three</FlexItem>
        <FlexItem bg="blue.50" flex="auto" minWidth={120}>Four</FlexItem>
        <FlexItem bg="blue.50" flex="auto" minWidth={120}>Five</FlexItem>
        <FlexItem bg="blue.50" flex="auto" minWidth={120}>Six</FlexItem>
      </Flex>
    </>
  );
}
```

### The `align` (or `alignItems`) property

image

```jsx
function Example() {
  const FlexItem = (props) => (
    <Box p=".75rem" mr="1rem" mb="1rem" {...props} />
  );

  const [value, setValue] = React.useState('stretch');

  return (
    <>
      <Box mb="1rem">
        <Button onClick={() => setValue('stretch')}>stretch</Button>
        <Button onClick={() => setValue('flex-start')}>flex-start</Button>
        <Button onClick={() => setValue('center')}>center</Button>
        <Button onClick={() => setValue('flex-end')}>flex-end</Button>
      </Box>
      <pre>
        {`<Flex align="${value}">`}
      </pre>
      <Flex align={value} height={240} bg="gray.20" p="1rem" pb={0} pr={0}>
        <FlexItem bg="blue.50" flex="auto">One</FlexItem>
        <FlexItem bg="blue.50" flex="auto">Two</FlexItem>
        <FlexItem bg="blue.50" flex="auto">Three</FlexItem>
      </Flex>
    </>
  );
}
```

### The `justify` (or `justifyContent`) property

image

```jsx
function Example() {
  const FlexItem = (props) => (
    <Box p=".75rem" mr="1rem" mb="1rem" {...props} />
  );

  const [value, setValue] = React.useState('start');

  return (
    <>
      <Box mb="1rem">
        <Button onClick={() => setValue('flex-start')}>flex-start</Button>
        <Button onClick={() => setValue('center')}>center</Button>
        <Button onClick={() => setValue('flex-end')}>flex-end</Button>
        <Button onClick={() => setValue('space-between')}>space-between</Button>
        <Button onClick={() => setValue('space-around')}>space-around</Button>
        <Button onClick={() => setValue('space-evenly')}>space-evenly</Button>
      </Box>
      <pre>
        {`<Flex justify="${value}">`}
      </pre>
      <Flex justify={value} bg="gray.20" p="1rem" pb={0} pr={0}>
        <FlexItem bg="blue.50">One</FlexItem>
        <FlexItem bg="blue.50">Two</FlexItem>
        <FlexItem bg="blue.50">Three</FlexItem>
      </Flex>
    </>
  );
}
```

### CODE REVIEW

File: packages/styled-docs/pages/flex.mdx [hunk 1]
```diff
@@ -1 +1,205 @@
 # Flex
+
+Flex is [Box](box) with `display: flex` and comes with helpful style shorthand props.
+
+## Import
+
+```js
+import Flex from '@trcolumn-reversemicro/react-styled-core/Flex';
```
Comment (austinhung81) [inline packages/styled-docs/pages/flex.mdx]: Is the "@trcolumn-reversemicro"  should be "@trendmicro"?

Comment (cheton) [reply packages/styled-docs/pages/flex.mdx]: 😱 DevOooooooooops.....

caused by `find & replace all` mistakes

**Comment by roth1002** [comment]
Resolved #19

File: packages/styled-docs/components/Main.jsx [hunk 2]
```diff
@@ -15,6 +17,14 @@ const Main = React.forwardRef(({ children, ...props }, ref) => {
       py=".75rem"
       backgroundColor={backgroundColor}
       minHeight="100vh"
+      css={css`
+        >:first-child {
+          margin-top: 0!important;
```
Comment (roth1002) [inline packages/styled-docs/components/Main.jsx:22]: important XD

Comment (roth1002) [reply packages/styled-docs/components/Main.jsx:22]: table就是麻煩XD

Comment (cheton) [reply packages/styled-docs/components/Main.jsx:22]: 這個是抄 GitHub 的，因為要套用到整個 markdown 文件第一層級的 elements，看來是不得以的做法

**Review by roth1002 — APPROVED** [review]

**Pull request merged**